src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonOutputFormatter.cs

```diff
@@ -82,7 +82,7 @@ public sealed override async Task WriteResponseBodyAsync(OutputFormatterWriteCon
                 await JsonSerializer.SerializeAsync(responseStream, context.Object, objectType, SerializerOptions, httpContext.RequestAborted);
                 await responseStream.FlushAsync(httpContext.RequestAborted);
             }
-            catch (OperationCanceledException) { }
+            catch (OperationCanceledException ex) when (ex.CancellationToken == httpContext.RequestAborted) { }
```

Comment:
Should we instead check

```suggestion
            catch (OperationCanceledException) when (httpContext.RequestAborted.IsCancellationRequested) { }
```